cli/flox-rust-sdk/src/providers/publish.rs

```diff
@@ -82,23 +88,81 @@ pub struct CheckedBuildMetadata {
     _private: (),
 }
 
+#[allow(async_fn_in_trait)]
+pub trait BinaryCache {
+    async fn upload(&self, path: &str) -> Result<(), PublishError>;
```

Comment:
suggestion: remove `async`
I dont see anything in either implementation that suggests that we need to do any async task, and async interfaces that we consume can be integrated using e.g. `pollster`